expr/funcs.go

```diff
@@ -49,18 +49,20 @@ func init() {
 		"alias":          {NewAlias, true},
 		"aliasByNode":    {NewAliasByNode, true},
 		"aliasSub":       {NewAliasSub, true},
-		"avg":            {NewAvgSeries, true},
-		"averageSeries":  {NewAvgSeries, true},
+		"avg":            {NewAggregateConstructor("average"), true},
+		"averageSeries":  {NewAggregateConstructor("average"), true},
```

Comment:
That's totally doable. I doubt it will have any real performance impact on this transient object, but it would help catch typo's at compile-time when new functions are added/changed.

Reply:
So, I implemented something, but now I'm not as sure about the benefit. I created a struct that holds a function and a name. I can use a set of structs, or else we move back to the interfaces. So, now the name and the struct can get out of sync (bad copy/pasta, etc). Can you give me more insight into what you would like?

Reply:
So, at compile time, a non-existent function can be caught. At test time, typo's / mismatch between name and function can be caught. 

At startup (specifically, `funcs::init()`) the function and name would be assigned (no lookup needed). 

Is this acceptable?